open_instruct/grpo_vllm_thread_ray_gtrl.py

```diff
@@ -642,15 +641,15 @@ def from_pretrained(
         # optim_params = get_optimizer_grouped_parameters(self.policy, weight_decay)
         # self.optimizer = AdamOptimizer(optim_params, lr=args.learning_rate)
         self.optimizer = torch.optim.AdamW(self.policy.parameters(), lr=args.learning_rate)
-        num_training_steps = args.num_training_steps * args.num_train_epochs * args.num_epochs
+        num_scheduler_steps = args.num_training_steps * args.num_epochs * args.num_mini_batches
```

Comment:
I guess the scheduler steps every ministep, not ministep*grad acc.

Reply:
This is because we do a gradient step at each mini batches (i.g., local_mini_batch_size)